### PR TITLE
Diamond & Hamming Exercism

### PR DESCRIPTION
**Pull request by breyana**
Only need to check out `diamond.js` and `hamming.js`

`*spec.js` holds the tests

### CODE REVIEW

File: exercism/accumulate/accumulate.spec.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 var accumulate = require('./accumulate');
 
-describe('accumulate()', function() {
+xdescribe('accumulate()', function() {
```
Comment (nodatall) [inline exercism/accumulate/accumulate.spec.js:3]: Why did you add x's before these tests? It makes them not run.

File: exercism/diamond/diamond.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+var Diamond = function() {};
+
+Diamond.prototype.makeDiamond = function(input) {
+  let letter = input.toUpperCase()
+let alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
```
Comment (nodatall) [inline exercism/diamond/diamond.js:5]: Indent everything inside `Diamond.prototype.makeDiamond = function(input) {`

File: exercism/diamond/diamond.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+let diamondLimit = []
+let diamondArray = []
+
+for (let i = 0; i < alphabet.length; i++) {
+  diamondLimit.push(alphabet[i])
+  if (alphabet[i] === letter) {
+    break;
+  }
+}
```
Comment (nodatall) [inline exercism/diamond/diamond.js:14]: This loop could replaced with a shorter and more elegant slice:
```js
diamondLimit = alphabet.slice(0, alphabet.indexOf(letter) + 1).split('')
```

File: exercism/diamond/diamond.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+var Diamond = function() {};
+
+Diamond.prototype.makeDiamond = function(input) {
+  let letter = input.toUpperCase()
+let alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+let diamondLimit = []
```
Comment (nodatall) [inline exercism/diamond/diamond.js:6]: `diamondLimit` doesn't express what this array is.  You could rename it to `diamondLetters`.

File: exercism/diamond/diamond.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    break;
+  }
+}
+
+let diamondSize = diamondLimit.length * 2 - 1
+
+for (let i = 0; i < diamondSize; i++) {
+  diamondArray.push(Array(diamondSize).fill(' '))
+}
+let leftLetterPlacement = diamondLimit.length - 1
```
Comment (nodatall) [inline exercism/diamond/diamond.js]: You could declare these at the same time:
```js
let leftLetterPlacement = rightLetterPlacement = diamondLimit.length - 1
```

File: exercism/diamond/diamond.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+let diamondArray = []
+
+for (let i = 0; i < alphabet.length; i++) {
+  diamondLimit.push(alphabet[i])
+  if (alphabet[i] === letter) {
+    break;
+  }
+}
+
+let diamondSize = diamondLimit.length * 2 - 1
```
Comment (nodatall) [inline exercism/diamond/diamond.js:16]: This is more like `diamondHeight`

File: exercism/diamond/diamond.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+for (let i = 0; i < diamondSize; i++) {
+  diamondArray.push(Array(diamondSize).fill(' '))
+}
+let leftLetterPlacement = diamondLimit.length - 1
+let rightLetterPlacement = diamondLimit.length - 1
+
+let i = 0
+let j = 0
+while (i < diamondArray.length/2) {
```
Comment (nodatall) [inline exercism/diamond/diamond.js:26]: Space for readability `diamondArray.length / 2)`

File: exercism/diamond/diamond.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  diamondArray[i][rightLetterPlacement] = diamondLimit[j]
+  diamondArray[i][leftLetterPlacement] = diamondLimit[j]
+  rightLetterPlacement--
+  leftLetterPlacement++
+  j++
+  i++
```
Comment (nodatall) [inline exercism/diamond/diamond.js:32]: `j` and `i` are the same in the scope of this while loop, no need to have two separate variables.

File: exercism/diamond/diamond.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+j = diamondLimit.length - 2
```
Comment (nodatall) [inline exercism/diamond/diamond.js:35]: Declare `j` here instead of on line 25

File: exercism/hamming/hamming.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
+var Hamming = function() {};
+
+Hamming.prototype.checkForSameLength = function(firstStrand, secondStrand) {
+  if (firstStrand.length === secondStrand.length) {
```
Comment (nodatall) [inline exercism/hamming/hamming.js:4]: Ternaries are cool: 
```
return firstStrand.length === secondStrand.length ? true : false
```

File: exercism/hamming/hamming.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+  if (this.checkForSameLength(firstStrand, secondStrand)) {
+    let firstArray = firstStrand.split('');
+    let secondArray = secondStrand.split('');
+    let hammingDistance = 0
+    for (var i = 0; i < firstArray.length; i++) {
+      if (firstArray[i] !== secondArray[i]) {
+        hammingDistance++
+      }
+    }
+    return hammingDistance
```
Comment (nodatall) [inline exercism/hamming/hamming.js:20]: Using a counter outside of a for loop is totally !rad. Use reduce instead:
```js
return firstStrand.split('').reduce( function (accumulator, currentChar, index) {
  return currentChar !== secondStrand[index] ? accumulator + 1 : accumulator
}, 0) 
```

**Comment by breyana** [comment]
Incorporated the changes in branch WedsExercism
